### PR TITLE
Python frontend: Class attributes

### PR DESCRIPTION
**Pull request by brcfarias**
This PR adds support for **class attributes** as described by Python documentation (https://docs.python.org/3/reference/compound_stmts.html#class-definitions):

"_Variables defined in the class definition are class attributes; they are shared by instances. Instance attributes can be set in a method with self.name = value. Both class and instance attributes are accessible through the notation “self.name”, and an instance attribute hides a class attribute with the same name when accessed in this way._"

Class attributes are modeled as static class members in C++. Thus, when a class attribute is identified, a symbol with an id in the format _filename@C@classname@varname_ is inserted into the context.

### CODE REVIEW

**Comment by brcfarias** [comment]
@rafaelsamenezes It looks like the CI failed because the source branch is only in my project fork.

**Review by fbrausse — CHANGES_REQUESTED** [review]
Instead of modifying the existing test case, could you please add a new one testing the new support for class attributes?

> @rafaelsamenezes It looks like the CI failed because the source branch is only in my project fork.

While that's true and we should allow PRs from forks (because it's standard procedure on Github), it seems that there are formatting changes. @brcfarias Could you double-check that everything touched here conforms to clang-format-11?

**Comment by lucasccordeiro** [comment]
@brcfarias: I agree with @fbrausse's comments. Could you please address them?

File: src/python-frontend/python_converter.h [hunk 1]
```diff
@@ -47,4 +49,8 @@ class python_converter
   std::string current_func_name;
   std::string current_class_name;
   exprt *ref_instance;
+  bool is_converting_lhs = false;
+
+  // Map object to list of instance attributes
+  std::map<std::string, std::vector<std::string>> instance_attr_map;
```
Comment (kunjsong01) [inline src/python-frontend/python_converter.h]: perhaps using unordered_map for better time complexity if the order doesn't matter?

**Review by kunjsong01 — CHANGES_REQUESTED** [review]
just a nitpick for minor performance improvement...

**Comment by brcfarias** [comment]
@fbrausse Thanks for the comments.

> Instead of modifying the existing test case, could you please add a new one testing the new support for class attributes?

I just updated the PR with a new regression test for class attributes.

> Could you double-check that everything touched here conforms to clang-format-11?

I was using clang-format-9 :grinning:

File: src/python-frontend/python_converter.h (hunk 1 above)
Comment (brcfarias) [reply src/python-frontend/python_converter.h]: Thanks @kunjsong01. Just changed the type to unordered_map.

**Review by lucasccordeiro — APPROVED** [review]

**Pull request merged**